### PR TITLE
[Serde](Variant) support arrow serialization for varint type

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

File: be/src/vec/data_types/serde/data_type_object_serde.h [hunk 1]
```diff
@@ -73,10 +73,7 @@ class DataTypeObjectSerDe : public DataTypeSerDe {
 
     void write_column_to_arrow(const IColumn& column, const NullMap* null_map,
                                arrow::ArrayBuilder* array_builder, int start, int end,
-                               const cctz::time_zone& ctz) const override {
-        throw doris::Exception(ErrorCode::NOT_IMPLEMENTED_ERROR,
-                               "write_column_to_arrow with type " + column.get_name());
-    }
+                               const cctz::time_zone& ctz) const override;
     void read_column_from_arrow(IColumn& column, const arrow::Array* arrow_array, int start,
```
Comment (amorynan) [inline be/src/vec/data_types/serde/data_type_object_serde.h:77]: dose we need to support read from arrow? for some spark situation : make string value to variant data type value into doris

Comment (eldenmoon) [reply be/src/vec/data_types/serde/data_type_object_serde.h:77]: currently this pr is to solve the connector problem for variant, for read from arrow we could use cast expr like cast(xxx as variant)

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Pull request merged**